app/queries/servers/custom_profile.ts

```diff
@@ -0,0 +1,177 @@
+    database: Database,
+    attributes: CustomProfileAttributeModel[] | null | undefined,
+    sortFn?: (a: CustomAttribute, b: CustomAttribute) => number,
+): Promise<CustomAttribute[]> => {
+    if (!attributes?.length) {
+        return [];
+    }
+
+    // We need to fetch field details for names and sorting
+    const fieldIds = attributes.map((attr) => attr.fieldId);
+    const fieldsQuery = await database.get<CustomProfileFieldModel>(CUSTOM_PROFILE_FIELD).query(
+        Q.where('id', Q.oneOf(fieldIds)),
+    ).fetch();
+
+    // Create a map of field IDs to names
+    const fieldsMap = new Map();
+    fieldsQuery.forEach((field) => {
+        fieldsMap.set(field.id, {
+            name: field.name,
+            sort_order: field.attrs?.sort_order || 0,
+        });
+    });
+
+    // Convert DB attributes to CustomAttribute format
+    const customAttrs = attributes.map((attr) => ({
+        id: attr.fieldId,
+        name: fieldsMap.get(attr.fieldId)?.name || attr.fieldId,
+        value: attr.value,
+        sort_order: fieldsMap.get(attr.fieldId)?.sort_order || 0,
+    }));
+
+    // Sort if a sort function is provided
+    return sortFn ? customAttrs.sort(sortFn) : customAttrs;
+};
+
+/**
+ * Query to fetch all custom profile fields
+ * @param database - The database instance
+ * @returns Query for custom profile fields
+ */
+export const queryCustomProfileFields = (database: Database) => {
+    return database.get<CustomProfileFieldModel>(CUSTOM_PROFILE_FIELD).query(Q.where('delete_at', 0));
+};
+
+/**
+ * Query to fetch custom profile attributes for a user
+ * @param database - The database instance
+ * @param userId - The user ID
+ * @returns Query for custom profile attributes
+ */
+export const queryCustomProfileAttributesByUserId = (database: Database, userId: string) => {
+    return database.get<CustomProfileAttributeModel>(CUSTOM_PROFILE_ATTRIBUTE).query(
+        Q.where('user_id', userId),
+    );
+};
+
+/**
+ * Query to fetch custom profile attributes by field ID
+ * @param database - The database instance
+ * @param fieldId - The field ID
+ * @returns Query for custom profile attributes
+ */
+export const queryCustomProfileAttributesByFieldId = (database: Database, fieldId: string) => {
+    return database.get<CustomProfileAttributeModel>(CUSTOM_PROFILE_ATTRIBUTE).query(
+        Q.where('field_id', fieldId),
+    );
+};
+
+/**
+ * Query to delete custom profile attributes by field ID
+ * @param database - The database instance
+ * @param fieldId - The field ID
+ * @returns Promise that resolves when the deletion is complete
+ */
+export const deleteCustomProfileAttributesByFieldId = async (database: Database, fieldId: string) => {
+    const attributes = await queryCustomProfileAttributesByFieldId(database, fieldId);
+
+    return database.write(async () => {
+        await Promise.all(attributes.map((attribute) => attribute.destroyPermanently()));
+    });
```

Comment:
Shouldn't this be an operator handler?

Also much faster if you prepare to destroy and then batch